### PR TITLE
fix: improve retry visibilty

### PR DESCRIPTION
**Pull request by johnnagro**
## Description

I am not getting the behavior i expect. This makes several changes: 

- It modifies a different error handler. I had modified the read error handler, where perhaps i should be doing the connection error handler because they treat the error state differently. The read error one also looks at the verb and status code and the exceptions i am interested in do not have a status.
- It adds a log line when retries are exhausted
- It explicitly sets the read and connect retry counts. The docs say the total takes precedence but I cannot find where the read and connect counts are set if not in the init.

https://github.com/urllib3/urllib3/blob/main/src/urllib3/util/retry.py#L460-L472

### CODE REVIEW

File: enterprise_catalog/apps/api_client/base_oauth_with_retry.py [hunk 1]
```diff
@@ -35,9 +35,18 @@ def increment(
...
+        logger.info(f'EnterpriseRetry retrying {method} to {url}, error={error}...')
         return super().increment(method, url, response, error, _pool, _stacktrace)
 
+    def is_exhausted(self) -> bool:
+        """
+        Are we out of retries? Adding logging.
+        """
+        super_result = super().is_exhausted()
+        if super_result:
+            logger.info('EnterpriseRetry retries exhausted.')
```
Comment (marlonkeating) [inline enterprise_catalog/apps/api_client/base_oauth_with_retry.py:47]: Does this give enough logging info for us to know the context?

File: enterprise_catalog/apps/api_client/base_oauth_with_retry.py [hunk 2]
```diff
@@ -56,6 +65,8 @@ def __init__(
         super().__init__(**kwargs)
         retry = EnterpriseRetry(
             total=max_retries,
+            read=max_retries,
+            connect=max_retries,
```
Comment (marlonkeating) [inline enterprise_catalog/apps/api_client/base_oauth_with_retry.py:69]: Where is max_retries defined?

**Review by marlonkeating — CHANGES_REQUESTED** [review]

File: enterprise_catalog/apps/api_client/base_oauth_with_retry.py (hunk 2 above)
Comment (iloveagent57) [reply enterprise_catalog/apps/api_client/base_oauth_with_retry.py:69]: it's a parameter to `__init__()`

File: enterprise_catalog/apps/api_client/base_oauth_with_retry.py (hunk 1 above)
Comment (iloveagent57) [reply enterprise_catalog/apps/api_client/base_oauth_with_retry.py:47]: I _think_ we'd get the task id in the logs along with this line, which is probably enough to go on?

**Review by iloveagent57 — APPROVED** [review]

**Review by marlonkeating — APPROVED** [review]

**Pull request merged**